src/OpenLoco/src/Drawing/DrawingContext.h

```diff
@@ -43,112 +43,100 @@ namespace OpenLoco::Drawing
+            uint16_t width,
             AdvancedColour colour,
             StringId stringId,
             const void* args = nullptr)
             = 0;
 
-        virtual void drawStringRightUnderline(
+        virtual void drawStringCentred(
             Gfx::RenderTarget& rt,
-            int16_t x,
-            int16_t y,
+            Ui::Point& origin,
             AdvancedColour colour,
             StringId stringId,
-            const void* args)
+            const void* args = nullptr)
             = 0;
 
-        virtual void drawStringLeftUnderline(
+        virtual void drawStringCentredClipped(
             Gfx::RenderTarget& rt,
-            int16_t x,
-            int16_t y,
+            Ui::Point& origin,
+            uint16_t width,
             AdvancedColour colour,
             StringId stringId,
             const void* args = nullptr)
             = 0;
 
-        virtual void drawStringCentred(
+        virtual void drawStringCentredRaw(
             Gfx::RenderTarget& rt,
-            int16_t x,
-            int16_t y,
+            Ui::Point& origin,
+            uint16_t linebreakCount,
             AdvancedColour colour,
-            StringId stringId,
-            const void* args = nullptr)
+            const char* wrappedStr)
             = 0;
 
-        virtual void drawStringCentredClipped(
+        virtual uint16_t drawStringCentredWrapped(
             Gfx::RenderTarget& rt,
-            int16_t x,
-            int16_t y,
-            int16_t width,
+            Ui::Point& origin,
+            uint16_t width,
             AdvancedColour colour,
             StringId stringId,
             const void* args = nullptr)
             = 0;
 
-        virtual uint16_t drawStringCentredWrapped(
+        virtual void drawStringRight(
             Gfx::RenderTarget& rt,
             Ui::Point& origin,
-            uint16_t width,
             AdvancedColour colour,
             StringId stringId,
             const void* args = nullptr)
             = 0;
 
-        virtual void drawStringCentredRaw(
+        virtual void drawStringRightUnderline(
             Gfx::RenderTarget& rt,
-            int16_t x,
-            int16_t y,
-            int16_t linebreakCount,
+            Ui::Point& origin,
             AdvancedColour colour,
-            const char* wrappedStr)
+            StringId stringId,
+            const void* args)
             = 0;
 
         virtual void drawStringYOffsets(Gfx::RenderTarget& rt, const Ui::Point& loc, AdvancedColour colour, const void* args, const int8_t* yOffsets) = 0;
 
-        virtual void drawStringTicker(Gfx::RenderTarget& rt, const Ui::Point& origin, StringId stringId, Colour colour, uint8_t numLinesToDisplay, uint16_t numCharactersToDisplay, uint16_t width) = 0;
+        virtual void drawStringTicker(Gfx::RenderTarget& rt, Ui::Point& origin, StringId stringId, Colour colour, uint8_t numLinesToDisplay, uint16_t numCharactersToDisplay, uint16_t width) = 0;
```

Comment:
what was the reason to remove `const`? I would have thought all of these draw methods would take a const reference

Reply:
We just had a nice discussion about this on Discord. It will be changed to `Point` instead of a ref.